core/src/main/java/org/apache/calcite/sql/fun/SqlDialectOperatorTableFactory.java

```diff
@@ -0,0 +1,247 @@
+  public static final SqlFunction RTRIM =
+      new SqlFunction("RTRIM", SqlKind.RTRIM,
+          ReturnTypes.cascade(ReturnTypes.ARG0, SqlTypeTransforms.TO_NULLABLE,
+              SqlTypeTransforms.TO_VARYING), null,
+          OperandTypes.STRING, SqlFunctionCategory.STRING);
+
+  /** Oracle's "SUBSTR(string, position [, substringLength ])" function.
+   *
+   * <p>It has similar semantics to standard SQL's
+   * {@link SqlStdOperatorTable#SUBSTRING} function but different syntax. */
+  @SqlDialect(dialects = {Dialect.ORACLE})
+  public static final SqlFunction SUBSTR =
+      new SqlFunction("SUBSTR", SqlKind.OTHER_FUNCTION,
+          ReturnTypes.ARG0_NULLABLE_VARYING, null, null,
+          SqlFunctionCategory.STRING);
+
+  /** The "GREATEST(value, value)" function. */
+  @SqlDialect(dialects = {Dialect.ORACLE})
+  public static final SqlFunction GREATEST =
+      new SqlFunction("GREATEST", SqlKind.GREATEST,
+          ReturnTypes.cascade(ReturnTypes.LEAST_RESTRICTIVE,
+              SqlTypeTransforms.TO_NULLABLE), null,
+          OperandTypes.SAME_VARIADIC, SqlFunctionCategory.SYSTEM);
+
+  /** The "LEAST(value, value)" function. */
+  @SqlDialect(dialects = {Dialect.ORACLE})
+  public static final SqlFunction LEAST =
+      new SqlFunction("LEAST", SqlKind.LEAST,
+          ReturnTypes.cascade(ReturnTypes.LEAST_RESTRICTIVE,
+              SqlTypeTransforms.TO_NULLABLE), null,
+          OperandTypes.SAME_VARIADIC, SqlFunctionCategory.SYSTEM);
+
+  /**
+   * The <code>TRANSLATE(<i>string_expr</i>, <i>search_chars</i>, <i>replacement_chars</i>)</code>
+   * function returns <i>string_expr</i> with all occurrences of each character in
+   * <i>search_chars</i> replaced by its corresponding character in <i>replacement_chars</i>.
+   *
+   * <p>It is not defined in the SQL standard, but occurs in Oracle and PostgreSQL.
+   */
+  @SqlDialect(dialects = {Dialect.ORACLE, Dialect.POSTGRES})
+  public static final SqlFunction TRANSLATE3 = new SqlTranslate3Function();
+
+  /**
+   * Returns the sql dialect operator table factory, creating it if necessary.
+   */
+  public static synchronized SqlDialectOperatorTableFactory instance() {
+    if (instance == null) {
+      // Creates and initializes this table factory.
+      instance = new SqlDialectOperatorTableFactory();
+      instance.init();
+    }
+    return instance;
+  }
+
+  //~ Methods ----------------------------------------------------------------
+
+  /**
+   * Performs post-constructor initialization of an operator table. It can't
+   * be part of the constructor, because the subclass constructor needs to
+   * complete first.
+   */
+  public final void init() {
+    // Initialize the dialect to SqlOperator mapping.
+    for (Dialect dialect : Dialect.values()) {
+      dialectSqlOperatorTableMapping.put(dialect, new ListSqlOperatorTable());
+    }
+
+    // Use reflection to register the expressions stored in public fields.
+    for (Field field : getClass().getFields()) {
```

Comment:
`Use` -> `Uses`?

Reply:
I think "Use" (imperative verb) is ok since this is code. "Uses" (indicative verb) is for javadoc comments.